File: my-folder/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/solution.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        words = sentence.split()
        n = len(words)

        ans = -1
        for i in range(n):
            word = words[i]
            if len(word)<len(searchWord):
                continue
            
            else:
                l = r = 0
                is_prefix = True
                while r<len(searchWord):
                    if word[l]==searchWord[r]:
                        l+=1
                        r+=1
                    else:
                        is_prefix = False
                        break
                if is_prefix:
                    ans = i+1
                    return ans
        
        return ans

File: my-folder/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/solution.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        vector<string> words;
        string currWord; 
        for (char character : sentence){
            if (character != ' '){
                currWord+=character;
            }
            else{
                if (!currWord.empty()){
                    words.push_back(currWord);
                    currWord = "";
                }
            }
        }
        if (!currWord.empty()){
            words.push_back(currWord);
        }

        for (int wordIndex = 0; wordIndex < words.size(); wordIndex++ )
        {
            if (words[wordIndex].length() >= searchWord.length()){
                bool isMatch = true;
                for (int charIndex = 0; charIndex < searchWord.length(); charIndex++)
                {
                    if (searchWord[charIndex]!=words[wordIndex][charIndex]){
                        isMatch = false;
                        break;
                    }
                }
                if (isMatch)
                {
                    return wordIndex+1;
                }
            }
        }
        return -1;

    }
};
